### PR TITLE
fix: Update overcommit ratio during live VM migration

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

During live migration of a VM from between hosts having different cgroup versions (cgroupv2 & cgroup), overcommit ratio is ignored.

This PR fixes the above issue.

#### Steps to reproduce
1. Set overcommit ratio to 2 for the cluster
2. Launch a VM
3. Check value of shares on the host
```bash
$ virsh dumpxml --domain i-2-9-VM | grep shares
    <shares>500</shares>
```
4. Live migrate the VM to another host which has a different cgroup than the existing host of the VM.
5. Live migrate the VM back to the original host. And check the shares again.
```bash
$ virsh dumpxml --domain i-2-9-VM | grep shares
    <shares>1000</shares>
```
6. Stopping and starting the VM on the same host will reset the value back to `500`.
This happens only when cgroup is different on the destination hosts during live migration. Possibly introduced in #8252 .

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

Not able to reproduce the issue after the patch.

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -1462,6 +1450,23 @@ public void orchestrateStart(final String vmUuid, final Map<VirtualMachineProfil
         }
     }
 
+    private void updateOverCommitRatioForVmProfile(VirtualMachineProfile vmProfile, long clusterId) {
+        final ClusterDetailsVO clusterDetailCpu = _clusterDetailsDao.findDetail(clusterId, VmDetailConstants.CPU_OVER_COMMIT_RATIO);
+        final ClusterDetailsVO clusterDetailRam = _clusterDetailsDao.findDetail(clusterId, VmDetailConstants.MEMORY_OVER_COMMIT_RATIO);
+
+        if (userVmDetailsDao.findDetail(vmProfile.getId(), VmDetailConstants.CPU_OVER_COMMIT_RATIO) == null &&
```
Comment (sureshanaparti) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:1457]: can do this lookup once?

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java [hunk 2]
```diff
@@ -1462,6 +1450,23 @@ public void orchestrateStart(final String vmUuid, final Map<VirtualMachineProfil
...
+                (Float.parseFloat(clusterDetailCpu.getValue()) > 1f || Float.parseFloat(clusterDetailRam.getValue()) > 1f)) {
+            userVmDetailsDao.addDetail(vmProfile.getId(), VmDetailConstants.CPU_OVER_COMMIT_RATIO, clusterDetailCpu.getValue(), true);
+            userVmDetailsDao.addDetail(vmProfile.getId(), VmDetailConstants.MEMORY_OVER_COMMIT_RATIO, clusterDetailRam.getValue(), true);
+        } else if (userVmDetailsDao.findDetail(vmProfile.getId(), VmDetailConstants.CPU_OVER_COMMIT_RATIO) != null) {
+            userVmDetailsDao.addDetail(vmProfile.getId(), VmDetailConstants.CPU_OVER_COMMIT_RATIO, clusterDetailCpu.getValue(), true);
+            userVmDetailsDao.addDetail(vmProfile.getId(), VmDetailConstants.MEMORY_OVER_COMMIT_RATIO, clusterDetailRam.getValue(), true);
```
Comment (sureshanaparti) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java:1463]: both cases, same values are set

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]
LGTM - didn't test it but wondering if it can have any side-effects?

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Review by vladimirpetrov — APPROVED** [review]
LGTM based on manual testing, tested on Oracle Linux 9.

**Comment by rohityadavcloud** [comment]
Merging this based on reviews, smoketests and manual QA.

**Pull request merged**